nsdl-c/sn_nsdl_lib.h

```diff
@@ -135,53 +135,41 @@ typedef enum sn_nsdl_resource_mode_ {
 } sn_nsdl_resource_mode_e;
 
 /**
- * \brief Resource registration parameters
+ * \brief Defines static parameters for the resource.
  */
-typedef struct sn_nsdl_resource_parameters_ {
-    unsigned int     observable:2;
-    unsigned int     registered:2;
-
+typedef struct sn_nsdl_static_resource_parameters_ {
+    int16_t     pathlen;                    /**< Address */ // Check type
+    uint16_t    resourcelen;                /**< 0 if dynamic resource, resource information in static resource */
     uint16_t    resource_type_len;
     uint16_t    interface_description_len;
-
-    uint16_t    coap_content_type;
-//    uint8_t     mime_content_type;
-
-    uint8_t     *resource_type_ptr;
-    uint8_t     *interface_description_ptr;
-
-} sn_nsdl_resource_parameters_s;
+    uint16_t    coap_content_type;          // This can be removed?
+    uint8_t     access:4;
```

Comment:
At some point this access could be moved to be after the pointers to save 1 byte of padding.